webapp/src/ts/training-card.guard.provider.ts

```diff
@@ -0,0 +1,47 @@
+import { ActivatedRouteSnapshot, CanDeactivate, RouterStateSnapshot } from '@angular/router';
+import { Injectable } from '@angular/core';
+import { Store } from '@ngrx/store';
+
+import { GlobalActions } from '@mm-actions/global';
+import { Selectors } from '@mm-selectors/index';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TrainingCardDeactivationGuardProvider implements CanDeactivate<any> {
+  private globalActions;
+  private trainingCardOpen = false;
+  private formId: string | null = null;
+
+  constructor(private store: Store) {
+    this.globalActions = new GlobalActions(store);
+  }
+
+  canDeactivate(
+    component: any,
+    currentRoute: ActivatedRouteSnapshot,
+    currentState: RouterStateSnapshot,
+    nextState: RouterStateSnapshot,
+  ) {
+
+    this.store
+      .select(Selectors.getTrainingCard)
+      .subscribe(trainingCard => {
+        this.trainingCardOpen = trainingCard.isOpen;
+        this.formId = trainingCard.formId;
+      }).unsubscribe();
+
+    if (this.trainingCardOpen) {
+      this.globalActions.setTrainingCard({
+        formId: this.formId,
+        isOpen: true,
```

Comment:
So far, these two properties seem to have the same values as in the state, and there is no need to set them in the Guard. Did you spot a weird behavior that maybe I missed? 
```suggestion
      this.globalActions.setTrainingCard({
```

Reply:
No I did not, that was an oversight. I agree the guard does not need to update those values.